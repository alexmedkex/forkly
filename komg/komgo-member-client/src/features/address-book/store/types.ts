import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
