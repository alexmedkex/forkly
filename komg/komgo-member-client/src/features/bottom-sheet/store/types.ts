import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
