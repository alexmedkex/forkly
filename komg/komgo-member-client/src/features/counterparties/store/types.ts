import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import { IX500Name } from '@komgo/types'

// Actions types
export enum CounterpartiesActionType {
  FETCH_CONNECTED_COUNTERPARTIES_REQUEST = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_REQUEST',
  FETCH_CONNECTED_COUNTERPARTIES_SUCCESS = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_SUCCESS',
  FETCH_CONNECTED_COUNTERPARTIES_FAILURE = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_FAILURE',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_REQUEST = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_REQUEST',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE',
  ADD_COUNTERPARTY_REQUEST = '@@counterparties/ADD_COUNTERPARTY_REQUEST',
  ADD_COUNTERPARTY_SUCCESS = '@@counterparties/ADD_COUNTERPARTY_SUCCESS',
  ADD_COUNTERPARTY_FAILURE = '@@counterparties/ADD_COUNTERPARTY_FAILURE',
  FETCH_COUNTERPARTY_REQ_REQUEST = '@@counterparties/FETCH_COUNTERPARTY_REQ_REQUEST',
  FETCH_COUNTERPARTY_REQ_SUCCESS = '@@counterparties/FETCH_COUNTERPARTY_REQ_SUCCESS',
  FETCH_COUNTERPARTY_REQ_FAILURE = '@@counterparties/FETCH_COUNTERPARTY_REQ_FAILURE',
  RESEND_COUNTERPARTY_REQUEST = '@@counterparties/RESEND_COUNTERPARTY_REQUEST',
  RESEND_COUNTERPARTY_SUCCESS = '@@counterparties/RESEND_COUNTERPARTY_SUCCESS',
  RESEND_COUNTERPARTY_FAILURE = '@@counterparties/RESEND_COUNTERPARTY_FAILURE',
  RESPONSE_ON_COUNTERPARTY_REQ_REQUEST = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_REQUEST',
  RESPONSE_ON_COUNTERPARTY_REQ_SUCCESS = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_SUCCESS',
  RESPONSE_ON_COUNTERPARTY_REQ_FAILURE = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_FAILURE',
  SEARCH_COUNTERPARTY = '@@counterparties/SEARCH_COUNTERPARTY',
  SORT_CONNECTED_COUNTERPARTIES = '@@counterparties/SORT_CONNECTED_COUNTERPARTIES',
  SET_ADD_COUNTERPARTIES = '@@counterparties/SET_ADD_COUNTERPARTIES',
  SET_COUNTERPARTY_MODAL = '@@counterparties/SET_COUNTERPARTY_MODAL',
  SET_REQUEST_ACTION = '@@counterparties/SET_REQUEST_ACTION',
  FETCH_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_REQUEST',
  FETCH_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_SUCCESS',
  FETCH_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_FAILURE',
  UPDATE_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_REQUEST',
  UPDATE_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_SUCCESS',
  UPDATE_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_FAILURE',
  CREATE_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_REQUEST',
  CREATE_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_SUCCESS',
  CREATE_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_FAILURE'
}

// State
export interface CounterpartiesStateFields {
  counterpartiesSearch: string
  counterparties: Counterparty[]
  counterpartiesSort: Sort
  notConnectedCounterpartySearch: string
  notConnectedCounterparties: NotConnectedCounterparty[]
  addCounterparties: string[]
  isAddModalOpen: boolean
  requestResponseActionStatus: boolean
  counterpartyRequest: CounterpartyRequest | null
  counterpartyProfiles: Map<string, CounterpartyProfile>
}

// Immutable state
export type CounterpartiesState = ImmutableMap<CounterpartiesStateFields>

export enum CouneterpartyStatus {
  COMPLETED = 'COMPLETED',
  PENDING = 'PENDING',
  WAITING = 'WAITING'
}

export const CouneterpartyStatusText: { [key: string]: string } = {
  [CouneterpartyStatus.COMPLETED]: 'Completed',
  [CouneterpartyStatus.PENDING]: 'Request pending',
  [CouneterpartyStatus.WAITING]: 'Action Required'
}

// Data types
export interface Counterparty {
  isFinancialInstitution: boolean
  staticId: string
  isMember: boolean
  x500Name: IX500Name
  covered: boolean
  status?: string
  timestamp?: Date
  profile?: CounterpartyProfile
}

export interface CounterpartyRequest extends Counterparty {
  requestId: string
}

export interface CompanyInfo {
  CN: string
  O: string
  C: string
  L: string
  STREET: string
  PC: string
}

export interface NotConnectedCounterparty {
  staticId: string
  isMember: boolean
  x500Name: CompanyInfo
  status?: CouneterpartyStatus
}

export interface SearchCounterpartyPayload {
  search: string
  typeCounterparty: 'counterpartiesSearch' | 'notConnectedCounterpartySearch'
}

export interface Sort {
  column: string
  order: 'ascending' | 'descending' | ''
}

export interface ModalPayload {
  name: 'isAddModalOpen'
  value: boolean
}

export interface CounterpartyProfile {
  id: string
  counterpartyId: string
  riskLevel: RiskLevel
  renewalDate: string
  managedById: string
}

export enum RiskLevel {
  unspecified = '',
  low = 'low',
  medium = 'medium',
  high = 'high'
}

// Actions
export interface FetchCounterpartiesSuccess extends Action {
  payload: Counterparty[]
  type: CounterpartiesActionType.FETCH_CONNECTED_COUNTERPARTIES_SUCCESS
}

export interface FetchCounterpartiesError extends Action {
  type: CounterpartiesActionType.FETCH_CONNECTED_COUNTERPARTIES_FAILURE
  payload: string
}

export interface FetchNotConnectedCounterpartiesSuccess extends Action {
  payload: NotConnectedCounterparty[]
  type: CounterpartiesActionType.FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS
}

export interface FetchNotConnectedCounterpartiesError extends Action {
  type: CounterpartiesActionType.FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE
  payload: string
}

export interface RequestFetchSuccess extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_REQ_SUCCESS
  payload: any
}

export interface SearchCounterparty extends Action {
  type: CounterpartiesActionType.SEARCH_COUNTERPARTY
  payload: SearchCounterpartyPayload
}

export interface SortConnectedCounterparties extends Action {
  type: CounterpartiesActionType.SORT_CONNECTED_COUNTERPARTIES
  payload: Sort
}

export interface AddCounterpartySuccess extends Action {
  type: CounterpartiesActionType.ADD_COUNTERPARTY_SUCCESS
  payload: string[]
}

export interface SetCounterpartyModal extends Action {
  type: CounterpartiesActionType.SET_COUNTERPARTY_MODAL
  payload: ModalPayload
}

export interface SetAddCounterparties extends Action {
  type: CounterpartiesActionType.SET_ADD_COUNTERPARTIES
  payload: string[]
}

export interface CounterPartyResponseAction extends Action {
  type: CounterpartiesActionType.SET_REQUEST_ACTION
  payload: {
    status: boolean
  }
}

export interface FetchCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

export interface FetchCounterpartyProfileFailure extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_PROFILE_FAILURE
  payload: {
    error: any
    counterpartyId: string
  }
}

export interface CreateCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.CREATE_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

export interface UpdateCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.UPDATE_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

// All actions
export type CounterpartiesAction =
  | FetchCounterpartiesSuccess
  | FetchCounterpartiesError
  | RequestFetchSuccess
  | SetCounterpartyModal
  | FetchNotConnectedCounterpartiesError
  | FetchNotConnectedCounterpartiesSuccess
  | SearchCounterparty
  | SortConnectedCounterparties
  | AddCounterpartySuccess
  | SetAddCounterparties
  | CounterPartyResponseAction
  | FetchCounterpartyProfileSuccess
  | FetchCounterpartyProfileFailure
  | CreateCounterpartyProfileSuccess
  | UpdateCounterpartyProfileSuccess
