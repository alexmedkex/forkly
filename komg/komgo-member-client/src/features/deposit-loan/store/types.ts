import { ImmutableMap } from '../../../utils/types'
import {
  IDepositLoanResponse,
  ISharedDepositLoan,
  ISaveDepositLoan,
  Currency,
  DepositLoanPeriod,
  ISaveSharedDepositLoan,
  IDisclosedDepositLoanSummary,
  IDisclosedDepositLoan,
  DepositLoanType
} from '@komgo/types'
import { Map, List } from 'immutable'
import { IRequestInformationForm } from '../../credit-line/store/types'

export enum DepositLoanActionType {
  FetchDepositsLoansRequest = '@@deposit-loan/FETCH_DEPOSITS_LOANS_REQUEST',
  FetchDepositsLoansSuccess = '@@deposit-loan/FETCH_DEPOSITS_LOANS_SUCCESS',
  FetchDepositsLoansFailure = '@@deposit-loan/FETCH_DEPOSITS_LOANS_FAILURE',
  RemoveDepositLoanRequest = '@@deposit-loan/REMOVE_DEPOSIT_LOAN_REQUEST',
  RemoveDepositLoanSuccess = '@@deposit-loan/REMOVE_DEPOSIT_LOAN_SUCCESS',
  RemoveDepositLoanFailure = '@@deposit-loan/REMOVE_DEPOSIT_LOAN_FAILURE',
  GetDepositLoanRequest = '@@deposit-loan/GET_DEPOSIT_LOAN_REQUEST',
  GetDepositLoanSuccess = '@@deposit-loan/GET_DEPOSIT_LOAN_SUCCESS',
  GetDepositLoanFailure = '@@deposit-loan/GET_DEPOSIT_LOAN_FAILURE',
  CreateDepositLoanRequest = '@@deposit-loan/CREATE_DEPOSIT_LOAN_REQUEST',
  CreateDepositLoanSuccess = '@@deposit-loan/CREATE_DEPOSIT_LOAN_SUCCESS',
  CreateDepositLoanFailure = '@@deposit-loan/CREATE_DEPOSIT_LOAN_FAILURE',
  EditDepositLoanRequest = '@@deposit-loan/EDIT_DEPOSIT_LOAN_REQUEST',
  EditDepositLoanSuccess = '@@deposit-loan/EDIT_DEPOSIT_LOAN_SUCCESS',
  EditDepositLoanFailure = '@@deposit-loan/EDIT_DEPOSIT_LOAN_FAILURE',
  FetchDisclosedDepositLoanSummariesRequest = '@@deposit-loan/FETCH_DISCLOSED_DEPOSIT_LOAN_SUMMARIES_REQUEST',
  FetchDisclosedDepositLoanSummariesSuccess = '@@deposit-loan/FETCH_DISCLOSED_DEPOSIT_LOAN_SUMMARIES_SUCCESS',
  FetchDisclosedDepositLoanSummariesFailure = '@@deposit-loan/FETCH_DISCLOSED_DEPOSIT_LOAN_SUMMARIES_FAILURE',
  FetchDisclosedDepositsLoansForCurrencyAndTenorRequest = '@@deposit-loan/FETCH_DISCLOSED_DEPOSITS_LOANS_FOR_CURRENCY_AND_TENOR_REQUEST',
  FetchDisclosedDepositsLoansForCurrencyAndTenorSuccess = '@@deposit-loan/FETCH_DISCLOSED_DEPOSITS_LOANS_FOR_CURRENCY_AND_TENOR_SUCCESS',
  FetchDisclosedDepositsLoansForCurrencyAndTenorFailure = '@@deposit-loan/FETCH_DISCLOSED_DEPOSITS_LOANS_FOR_CURRENCY_AND_TENOR_FAILURE',
  CreateReqDepositLoanInformationRequest = '@@deposit-loan/CREATE_REQ_DEPOSIT_LOAN_INFO_REQUEST',
  CreateReqDepositLoanInformationSuccess = '@@deposit-loan/CREATE_REQ_DEPOSIT_LOAN_INFO_SUCCESS',
  CreateReqDepositLoanInformationFailure = '@@deposit-loan/CREATE_REQ_DEPOSIT_LOAN_INFO_FAILURE',
  FetchReqsDepositLoanRequest = '@@deposit-loan/FETCH_REQS_DEPOSIT_LOAN_REQUEST',
  FetchReqsDepositLoanSuccess = '@@deposit-loan/FETCH_REQS_DEPOSIT_LOAN_SUCCESS',
  FetchReqsDepositLoanFailure = '@@deposit-loan/FETCH_REQS_DEPOSIT_LOAN_FAILURE',
  DeclineReqsDepositLoanRequest = '@@deposit-loan/DECLINE_REQS_DEPOSIT_LOAN_REQUEST',
  DeclineReqsDepositLoanSuccess = '@@deposit-loan/DECLINE_REQS_DEPOSIT_LOAN_SUCCESS',
  DeclineReqsDepositLoanFailure = '@@deposit-loan/DECLINE_REQS_DEPOSIT_LOAN_FAILURE'
}

export enum CreditAppetiteDepositLoanFeature {
  Deposit = 'deposit',
  Loan = 'loan'
}

export interface ICurrencyAndTenor {
  currency: Currency
  period: DepositLoanPeriod
  periodDuration?: number
}

export interface ICurrencyAndTenorOption {
  value: string
  text: string
  content: string
}

export interface IExtendedSharedWith extends ISharedDepositLoan {
  sharedWithCompanyName?: string
}

export interface IExtendedDepositLoanResponse extends IDepositLoanResponse {
  currencyAndTenor?: string
  sharedWith: IExtendedSharedWith[]
}

export interface ISharedDepositLoanForm extends ISaveSharedDepositLoan {
  sharedWithCompanyName?: string
  staticId?: string
  updatedAt?: string
  requestStaticId?: string
}

export interface IExtendedDisclosedDepositLoanSummary extends IDisclosedDepositLoanSummary {
  currencyAndTenor: string
}

export interface DepositLoanDetailsQuery {
  currency: Currency
  period: DepositLoanPeriod
  periodDuration?: number
}

export interface IDepositLoanForm extends ISaveDepositLoan {
  staticId?: string
  currencyAndTenor?: string
  pricingUpdatedAt?: string
  sharedWith: ISharedDepositLoanForm[]
}

export interface IExtendedDisclosedDepositLoan extends IDisclosedDepositLoan {
  companyName: string
  companyLocation: string
}

export interface IRequestDepositLoanInformationForm extends IRequestInformationForm {
  type: DepositLoanType
}

// TODO: this should be in komgo types
export interface ICreateDepositLoanRequest {
  companyIds: string[]
  comment: string
  currency: Currency
  period: DepositLoanPeriod
  periodDuration?: number
  type: DepositLoanType
}

export enum RequestType {
  Received = 'received',
  Requested = 'requested'
}

// TODO: load this from komgo types !!!!
export enum DepositLoanRequestType {
  Requested = 'REQUESTED',
  Received = 'RECEIVED'
}

export enum DepositLoanRequestStatus {
  Pending = 'PENDING',
  Declined = 'DECLINED',
  Disclosed = 'DISCLOSED'
}
export interface IDepositLoanRequestDocument {
  staticId: string

  requestType: DepositLoanRequestType

  type: DepositLoanType
  currency: Currency
  period: DepositLoanPeriod
  periodDuration?: number
  comment: string
  companyStaticId: string

  // TODO Create same one
  status: DepositLoanRequestStatus

  createdAt: Date
  updatedAt: Date
  deletedAt?: Date
}

export interface IExtendRequestDepositLoan extends IDepositLoanRequestDocument {
  companyName: string
  currencyAndTenor: string
  currencyAndTenorStringValue: string
}

export interface DepoistLoanProperties {
  byId: Map<string, IDepositLoanResponse>
  summaries: List<IDisclosedDepositLoanSummary>
  disclosedById: Map<string, IDisclosedDepositLoan>
  requestsById: Map<string, IDepositLoanRequestDocument>
}

export interface DepositLoanProperties {
  [CreditAppetiteDepositLoanFeature.Deposit]: ImmutableMap<DepoistLoanProperties>
  [CreditAppetiteDepositLoanFeature.Loan]: ImmutableMap<DepoistLoanProperties>
}

export type DepositLoanState = ImmutableMap<DepositLoanProperties>
