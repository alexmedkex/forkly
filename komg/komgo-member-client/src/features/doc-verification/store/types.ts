import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'

export enum DocumentVerificationActionType {
  GET_SESSION_REQUEST = '@@licenses/GET_SESSION_REQUEST',
  GET_SESSION_SUCCESS = '@@licenses/GET_SESSION_SUCCESS',
  GET_SESSION_FAILURE = '@@licenses/GET_SESSION_FAILURE',
  VERIFY_DOCUMENT_REQUEST = '@@licenses/VERIFY_DOCUMENT_REQUEST',
  VERIFY_DOCUMENT_SUCCESS = '@@licenses/VERIFY_DOCUMENT_SUCCESS',
  VERIFY_DOCUMENT_FAILURE = '@@licenses/VERIFY_DOCUMENT_FAILURE'
}

export interface DocVerificationStateProperties {
  registeredAt: number
  companyName: string
  metadataHash: string
}

export type DocVerificationState = ImmutableMap<DocVerificationStateProperties>

export interface GetSessionRequest extends Action {
  type: DocumentVerificationActionType.GET_SESSION_REQUEST
}

export interface GetSessionSuccess extends Action {
  type: DocumentVerificationActionType.GET_SESSION_SUCCESS
  payload: { metadataHash: string }
}

export interface GetSessionFailure extends Action {
  type: DocumentVerificationActionType.GET_SESSION_FAILURE
  payload: string
}

export interface VerifyDocumentRequest extends Action {
  type: DocumentVerificationActionType.VERIFY_DOCUMENT_REQUEST
}

export interface VerifyDocumentSuccess extends Action {
  type: DocumentVerificationActionType.VERIFY_DOCUMENT_SUCCESS
  payload: { registeredAt: number; companyName: string }
}

export interface VerifyDocumentFailure extends Action {
  type: DocumentVerificationActionType.VERIFY_DOCUMENT_FAILURE
  payload: string
}

export type DocumentVerificationActions =
  | GetSessionRequest
  | GetSessionSuccess
  | GetSessionFailure
  | VerifyDocumentRequest
  | VerifyDocumentSuccess
  | VerifyDocumentFailure
