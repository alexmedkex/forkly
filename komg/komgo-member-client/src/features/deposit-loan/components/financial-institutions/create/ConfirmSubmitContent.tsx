import * as React from 'react'

import { IDepositLoanForm, IExtendRequestDepositLoan } from '../../../store/types'
import styled from 'styled-components'

interface IProps {
  depositLoan: IDepositLoanForm
  isEdit: boolean
  requests?: IExtendRequestDepositLoan[]
}

const ConfirmSubmitContent: React.FC<IProps> = (props: IProps) => {
  const { depositLoan, isEdit, requests } = props
  const { sharedWith } = depositLoan

  const withoutCounterparties = `Are you sure you want to add this currency and tenor to your list of currency and tenors?`
  const withCounterparties = `Are you sure you want to add this currency and tenor and share the selected information with the counterparties that have been added in "visibility to counterparties" ?`

  const editWithoutCounterparties = `Are you sure you want to edit this currency and tenor?`
  const editWithCounterparties = `Are you sure you want to edit this currency and tenor? All counterparties for whom you have modified information to which they have visibility will be notified. `

  const rejectedRequests = requests
    ? requests.filter(
        request => !depositLoan.sharedWith.find(shared => shared.sharedWithStaticId === request.companyStaticId)
      )
    : []

  const getDecliningText = () => {
    if (rejectedRequests.length) {
      return (
        <React.Fragment>
          <p data-test-id="declining-text">
            Counterparties for whom you have decided to disclose new information will be notified.
            <br />
            Couonterparties who have requested information and for whom you have decided not to diclose information will
            notified that their request is rejected. The counterparties concerned are:
          </p>
          {getListOfRejectedSellers()}
        </React.Fragment>
      )
    }
    return null
  }

  const getListOfRejectedSellers = () => {
    if (rejectedRequests.length) {
      return (
        <SellerList data-test-id="declined-companies">
          {rejectedRequests.map(request => (
            <li data-test-id="declined-company" key={request.companyStaticId}>
              {request.companyName}
            </li>
          ))}
        </SellerList>
      )
    }
    return null
  }

  if (isEdit) {
    return (
      <React.Fragment>
        <p>{sharedWith.length ? editWithCounterparties : editWithoutCounterparties}</p>
        {getDecliningText()}
      </React.Fragment>
    )
  }

  return (
    <React.Fragment>
      <p>{sharedWith.length ? withCounterparties : withoutCounterparties}</p>
      {getDecliningText()}
    </React.Fragment>
  )
}

const SellerList = styled.ul`
  font-weight: bold;
  padding-left: 15px;
`

export default ConfirmSubmitContent
