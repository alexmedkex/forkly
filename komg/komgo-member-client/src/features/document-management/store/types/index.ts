export * from './product'

export * from './category'

export * from './document-type'

export * from './document'

export * from './request'

export * from './template'

export * from './request-template'

export * from './request'

export * from './state'

export * from './modals'

export interface HasName {
  name: string
}

export interface HasId {
  id: string
}

export type Diff<T, U> = T extends U ? never : T // Remove types from T that are assignable to U
export type Filter<T, U> = T extends U ? T : never // Remove types from T that are not assignable to U
