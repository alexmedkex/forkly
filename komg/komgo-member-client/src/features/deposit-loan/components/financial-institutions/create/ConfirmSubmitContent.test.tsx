import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { buildFakeDepositLoanResponse } from '@komgo/types'

import ConfirmSubmitContent from './ConfirmSubmitContent'

describe('ConfirmSubmitContent', () => {
  const depositWithShared = buildFakeDepositLoanResponse()
  const depositWithoutShared = { ...depositWithShared, sharedWith: [] }

  it('should match default snaptshout edit=false, sharedWith exists', () => {
    expect(
      renderer.create(<ConfirmSubmitContent isEdit={false} depositLoan={depositWithShared} />).toJSON()
    ).toMatchSnapshot()
  })

  it('should match default snaptshout edit=false, sharedWith not exists', () => {
    expect(
      renderer.create(<ConfirmSubmitContent isEdit={false} depositLoan={depositWithoutShared} />).toJSON()
    ).toMatchSnapshot()
  })

  it('should match default snaptshout edit=true, sharedWith exists', () => {
    expect(
      renderer.create(<ConfirmSubmitContent isEdit={true} depositLoan={depositWithShared} />).toJSON()
    ).toMatchSnapshot()
  })

  it('should match default snaptshout edit=true, sharedWith not exists', () => {
    expect(
      renderer.create(<ConfirmSubmitContent isEdit={true} depositLoan={depositWithoutShared} />).toJSON()
    ).toMatchSnapshot()
  })
})
