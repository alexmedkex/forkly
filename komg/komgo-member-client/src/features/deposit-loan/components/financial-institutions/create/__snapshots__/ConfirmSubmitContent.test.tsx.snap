// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmSubmitContent should match default snaptshout edit=false, sharedWith exists 1`] = `
<p>
  Are you sure you want to add this currency and tenor and share the selected information with the counterparties that have been added in "visibility to counterparties" ?
</p>
`;

exports[`ConfirmSubmitContent should match default snaptshout edit=false, sharedWith not exists 1`] = `
<p>
  Are you sure you want to add this currency and tenor to your list of currency and tenors?
</p>
`;

exports[`ConfirmSubmitContent should match default snaptshout edit=true, sharedWith exists 1`] = `
<p>
  Are you sure you want to edit this currency and tenor? All counterparties for whom you have modified information to which they have visibility will be notified. 
</p>
`;

exports[`ConfirmSubmitContent should match default snaptshout edit=true, sharedWith not exists 1`] = `
<p>
  Are you sure you want to edit this currency and tenor?
</p>
`;
